cpp/src/fil/fil.cu

```diff
@@ -751,11 +754,24 @@ void tl2fil_sparse(std::vector<int>* ptrees, std::vector<fil_node_t>* pnodes,
   tl2fil_common(params, model, tl_params);
   tl2fil_sparse_check_t<fil_node_t>::check(model);
 
+  size_t num_trees = model.trees.size();
+
+  ptrees->reserve(num_trees);
```

Comment:
A slightly shorter version which avoids `push_back()` (feel free to use if you prefer it):

```
auto& trees = *ptrees;
trees.resize(num_trees);
trees[0] = 0;
for (size_t i = 1; i < num_trees; ++i) {
  trees[i] = trees[i - 1] + model.trees[i].num_nodes;
}
```

Reply:
I'm not sure why we'd want to avoid `push_back` here? To me, `reserve` plus `push_back` clearly signals the semantic intent of iteratively building a fixed number of elements, and then you also don't even need to think about whether there is a non-trivial initialization cost associated with your default elements. Unless there's something else that I'm not considering?

The one thing I will tweak here is shifting from `(*ptrees)[i]` to `ptrees->back()`. I think that reads a little more clearly if one were approaching this code cold.

Reply:
Either way is fine for me.